nixos/modules/config/system-path.nix

```diff
@@ -143,7 +257,10 @@ in
 
   config = {
 
-    environment.systemPackages = requiredPackages ++ config.environment.defaultPackages;
+    environment.systemPackages = lib.mkMerge [
+      requiredPackages
+      config.environment.defaultPackages
+    ];
```

Comment:
`defaultPackages` is exactly where this change shines, because the only reason `defaultPackages` was introduces is to be able to turn off packages installed by default! The default packages should definitely be changed to use the new attribute set definitions, so that its definitions can be overridden.